### PR TITLE
chore(ts): WIP - Convert utils

### PR DESCRIPTION
**Pull request by asturur**
This will be a partial conversion of utils, i will stop at the misc file and review/merge.
All together is a mess, but luckily we can also swap the imports at that point.

### CODE REVIEW

File: src/util/misc/radiansDegreesConversion.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+export const degreesToRadians = (degrees: TDegree): TRadian => (degrees * PiBy180) as TRadian;
+
+/**
+ * Transforms radians to degrees.
+ * @static
+ * @memberOf fabric.util
+ * @param {TRadian} radians value in radians
+ * @return {TDegree} value in degrees
+ */
+export const radiansToDegrees = (radians: TRadian): TDegree => (radians / PiBy180) as TDegree;
```
Comment (ShaMan123) [inline src/util/misc/radiansDegreesConversion.ts:20]: beautiful

File: src/util/misc/sin.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * @param {TRadian} angle the angle
+ * @return {Number} the sin value for angle.
+ */
+export const sin = (angle: TRadian): number => {
+  if (angle === 0) { return 0; }
+  const angleSlice = angle / halfPI;
+  let value = 1;
+  if (angle < 0) {
+    // sin(-a) = -sin(a)
+    value = -1;
```
Comment (ShaMan123) [inline src/util/misc/sin.ts:19]: make a const?
```suggestion
    value = Math.sign(angle);
```

Comment (asturur) [reply src/util/misc/sin.ts:19]: i always forget about Math.sign

**Comment by asturur** [comment]
i know what i broke, i ll try to fix it today.
I messed up the imports/exports

**Comment by asturur** [comment]
@ShaMan123 i had to revert a couple of imports.
The color class importing some util and the animate color util importing the color class was creating a circular dependency.

File: src/util/internals/removeFromArray.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Removes value from an array.
+ * Presence of value (and its position in an array) is determined via `Array.prototype.indexOf`
+ * @static
+ * @memberOf fabric.util
+ * @param {Array} array
+ * @param {*} value
+ * @return {Array} original array
+ */
+export const removeFromArray = (array: any[], value: any): any[] => {
```
Comment (asturur) [inline src/util/internals/removeFromArray.ts:10]: i m sure there is a way to tell Typescript that the returned array is the same type of the input array

**Comment by asturur** [comment]
Ok something broke with the import system and i had to patch it. I have no idea why.

File: src/util/internals/removeFromArray.ts (hunk 3 above)
Comment (ShaMan123) [reply src/util/internals/removeFromArray.ts:10]: Generic types

```suggestion
export function removeFromArray<T>(array: T[], value: T): T[] {
```

File: src/mixins/observable.mixin.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 //@ts-nocheck
 (function(global) {
   var fabric = global.fabric;
+  console.log(fabric)
```
Comment (ShaMan123) [inline src/mixins/observable.mixin.ts:4]: Hi!

**Review by ShaMan123 — COMMENTED** [review]
Regarding the circular import
I have seen it as well and managed to fix it in a different way without reverting
Try importing utils directly and not from the `util` index OR make `animate_color` or whatever a module (even as a hack).
That should fix it.
rollup seems to be a decent tool

File: src/mixins/observable.mixin.ts (hunk 4 above)
Comment (asturur) [reply src/mixins/observable.mixin.ts:4]: yeah i was investigating why fabric was undefined

File: src/util/internals/removeFromArray.ts (hunk 3 above)
Comment (asturur) [reply src/util/internals/removeFromArray.ts:10]: and of course T can be a mix of types. ( even if in our case often isn't )

Comment (asturur) [reply src/util/internals/removeFromArray.ts:10]: fat arrows aren't good for TS as functions?

**Comment by ShaMan123** [comment]
Can't be generic i thinl

בתאריך יום ד׳, 10 באוג׳ 2022, 19:28, מאת Andrea Bogazzi ‏<
***@***.***>:

File: src/util/misc/vectors.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+/**
+ * Creates a vetor from points represented as a point
+ * @static
+ * @memberOf fabric.util
+ *
+ * @param {Point} from
+ * @param {Point} to
+ * @returns {Point} vector
+ */
+export const createVector = (from: IPoint, to: IPoint): Point => new Point(to).subtract(from);
```
Comment (asturur) [inline src/util/misc/vectors.ts:32]: i don't get this. if the type was Point , way we changed it to interface? we expect devs to use points or objects with x/y?

Comment (ShaMan123) [reply src/util/misc/vectors.ts:32]: You are right
I committed it and felt the same
Reverr this 
And safeguard createVector instead

**Comment by asturur** [comment]
note for later self: `const foo = <T,>(x: T) => x;`

**Pull request merged**